src/gsy_e/models/strategy/state/storage_state.py

```diff
@@ -304,6 +357,17 @@ def market_cycle(self, past_time_slot, current_time_slot: DateTime,
             for energy_type in self._used_storage_share:
                 self.time_series_ess_share[past_time_slot][energy_type.origin] += energy_type.value
 
+    def _apply_losses_at_market_cycle(self, sold_energy_kWh: float, bought_energy_kWh: float):
+
+        charging_loss_kWh = bought_energy_kWh * self.losses.charging_loss_percent
+        discharging_loss_kWh = sold_energy_kWh * self.losses.discharging_loss_percent
+        self_discharging_kWh = (
+            self.losses.self_discharge_per_day_kWh * GlobalConfig.slot_length.total_days()
+        )
+        total_loss_kWh = charging_loss_kWh + discharging_loss_kWh + self_discharging_kWh
```

Comment:
Admittedly an overkill, but do you think that it makes sense to also keep the total_loss_kWh in the results for debugging purposes? On second thought, better to omit it and add it if necessary in the future.

Reply:
I think it is not an overkill and indeed interesting to know if you run storages with losses. Let me add it to the storage CSV file.

Reply:
I did add it here: https://github.com/gridsingularity/gsy-e/pull/1799/commits/3c11349d9e08f9fef7ea69b04f20cf990fe3a7d8
However, there is a corner case for that last market slot. We only calculate the losses for the past market slot in the market_cycle event method. But after the last market slot this is not triggered, hence the losses are not calculated. I gave it some time to fix, but could not find a solution for this. As you named this an overkill, I left it like this now. Let us deal with the last market slot once we need it. OK?

Reply:
Agreed and thanks!